packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -179,6 +179,42 @@ function parseIfCfnIntrinsic(object: any): any {
       const value = parseCfnValueToCdkValue(object[key]);
       return Fn.join(value[0], value[1]);
     }
+    case 'Fn::Cidr': {
+      const value = parseCfnValueToCdkValue(object[key]);
+
+      if (value.length === 2) {
+        return Fn.cidr(value[0], value[1]);
+      }
```

Comment:
Since we've empirically established that CloudFormation docs are lying here, let's just kill this `if`, and require 3 arguments.